lib/chef/knife/winrm_knife_base.rb

```diff
@@ -122,11 +122,12 @@ def relay_winrm_command(command)
 
             queue = Queue.new
             @winrm_sessions.each { |s| queue << s }
+            number_of_session = config[:concurrency] ? locate_config_value(:concurrency) : @name_args[0].split(" ").count || queue.size
             # These nils will kill the Threads once no more sessions are left
-            locate_config_value(:concurrency).times { queue << nil }
+            number_of_session.times { queue << nil }
```

Comment:
Array's default would anyway be nil, may be we can tweak the upcoming iteration logic.
Still if we want this, `queue += [nil] * number_of_sessions` would be a faster and nicer choice.
A sample Benchmark comparison with 30000 iterations:
Current: 6.8137
Suggested: 0.3112

Reply:
My bad, `queue` is an object of class [`Queue`](https://ruby-doc.org/core-2.5.0/Queue.html). This is not an array.